qutebrowser/browser/commands.py

```diff
@@ -245,37 +247,60 @@ def openurl(self, url=None, implicit=False,
                       clicking on a link).
             count: The tab index to open the URL in, or None.
         """
+        force_search = False
         if url is None:
             if tab or bg or window:
-                url = config.get('general', 'default-page')
+                urls = [config.get('general', 'default-page')]
             else:
                 raise cmdexc.CommandError("No URL given, but -t/-b/-w is not "
                                           "set!")
         else:
-            try:
-                url = objreg.get('quickmark-manager').get(url)
-            except urlmarks.Error:
-                try:
-                    url = urlutils.fuzzy_url(url)
-                except urlutils.InvalidUrlError as e:
-                    # We don't use cmdexc.CommandError here as this can be
-                    # called async from edit_url
-                    message.error(self._win_id, str(e))
-                    return
-        if tab or bg or window:
-            self._open(url, tab, bg, window, not implicit)
-        else:
-            curtab = self._cntwidget(count)
-            if curtab is None:
-                if count is None:
-                    # We want to open a URL in the current tab, but none exists
-                    # yet.
-                    self._tabbed_browser.tabopen(url)
-                else:
-                    # Explicit count with a tab that doesn't exist.
-                    return
+            urllist = [u for u in url.split('\n') if u.strip()]
+            if (len(urllist) > 1 and
+                    any(not urlutils.is_url(u) and
```

Comment:
It was thought as an heuristic. If the first line is an url probably all the lines contain an url otherwise the test is considered as an unique search query.